### ISSUE TITLE
inconsistent local/global assignment

### ISSUE THREAD

**Issue by u-fischer** [loopspace/tikzmark#12]
This here

~~~~
\RequirePackage[check-declarations,enable-debug]{expl3}
\documentclass{article}
\usepackage{tikz}
\usetikzlibrary{tikzmark}
\begin{document}
\tikzmark{blub}abc
\end{document}
~~~~

errors with 

~~~~
! LaTeX3 Error: Inconsistent local/global assignment

For immediate help type H <return>.
 ...                                              
                                                  
l.506 \bool_set_true:N \g__sn_preamble_bool
                                           
? h

This is a coding error.

Local assignment to a global variable '\g__sn_preamble_bool'.

? x
~~~~

as the library doesn't use `\bool_gset` when initializing the boolean:

~~~~
\bool_new:N \g__sn_preamble_bool
\bool_set_true:N \g__sn_preamble_bool
~~~~

**Comment by loopspace** [comment]
Bother.  I thought I'd caught all of those.  Thanks for finding it.

**Comment by loopspace** [comment]
Fixed in c2624af
